### PR TITLE
mds: update mdlog perf counters during replay

### PR DESCRIPTION
**Pull request by batrick**
Fixes: https://tracker.ceph.com/issues/61864

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/mds/MDLog.cc [hunk 1]
```diff
@@ -1432,6 +1433,8 @@ void MDLog::_replay_thread()
     }
 
     logger->set(l_mdl_rdpos, pos);
+    logger->set(l_mdl_expos, journaler->get_expire_pos());
+    logger->set(l_mdl_wrpos, journaler->get_write_pos());
```
Comment (lxbsz) [inline src/mds/MDLog.cc:1437]: Should we update these earlier ? What if the replay loop takes a long time ? Then in the perf dump we still couldn't see it ?

Comment (vshankar) [reply src/mds/MDLog.cc:1437]: Can these be updated +once_ before we enter the replay loop and then the relevant fields in the loop itself?

Comment (batrick) [reply src/mds/MDLog.cc:1437]: > Should we update these earlier ? What if the replay loop takes a long time ? Then in the perf dump we still couldn't see it ?

I think that's an issue only if the log event's replay gets stuck for the **first** log event.

Comment (batrick) [reply src/mds/MDLog.cc:1437]: > Can these be updated +once_ before we enter the replay loop and then the relevant fields in the loop itself?

I think I was just trying to keep these neat alongside the `rdpos` updates. The other concern is `standby-replay` daemon replay where the journaler write/expire pos is always changing. However, the standby-replay daemon should break out of this loop regularly anyway.

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test api

File: src/mds/MDLog.cc (hunk 1 above)
Comment (vshankar) [reply src/mds/MDLog.cc:1437]: Fair enough.

**Comment by vshankar** [comment]
Running into lots of "failed to fetch package" issues with the run such as - https://pulpito.ceph.com/vshankar-2023-07-25_11:25:54-fs-wip-vshankar-testing-20230725.053049-testing-default-smithi/7350458/

The kernel build are available. @lxbsz suspects bug in teuthology. Will keep you updated.

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#28-JULY-2023

**Pull request merged**